kpops/components/base_components/kafka_app.py

```diff
@@ -36,21 +36,44 @@ class KafkaAppConfig(KubernetesAppConfig):
 
 
 class KafkaApp(KubernetesApp):
-    """
-    Base component for Kafka-based components.
+    """Base component for Kafka-based components.
+
     Producer or streaming apps should inherit from this class.
+
+    :param type: Component type, defaults to "kafka-app"
+    :type type: str, optional
+    :param schema_type: Used for schema generation, same as :param:`type`,
+        defaults to "kafka-app"
+    :type schema_type: Literal["kafka-app"], optional
+    :param app: Application-specific settings
+    :type app: KafkaAppConfig
+    :param repo_config: Configuration of the Helm chart repo to be used for
+        deploying the component,
+        defaults to HelmRepoConfig(repository_name="bakdata-streams-bootstrap", url="https://bakdata.github.io/streams-bootstrap/")
+    :type repo_config: HelmRepoConfig, None, optional
+    :param version: Helm chart version, defaults to "2.9.0"
+    :type version: str, optional
     """
 
-    type: str = "kafka-app"
+    type: str = Field(default="kafka-app", description="Component type")
     schema_type: Literal["kafka-app"] = Field(  # type: ignore[assignment]
-        default="kafka-app", exclude=True
+        default="kafka-app",
+        title="Component type",
+        description=__doc__.partition(":param")[0].strip(),
```

Comment:
I could write a function that takes a class attribute name and returns the description from the docstring to further remove the repetition, but this could be considered going a little overboard. What do you think?

Reply:
interesting, I had no idea this is possible
yeah, this sounds like a good idea if you're using this in multiple places

Reply:
Welll, I didn't know if it was either :D
Ended up doing it a bit differently